packages/flutter_tools/lib/src/dart/generate_synthetic_packages.dart

```diff
@@ -51,19 +52,19 @@ Future<void> generateLocalizationsSyntheticPackage({
     if (isSyntheticL10nPackage == false) {
       return;
     }
-  } else if (!environment.useImplicitPubspecResolution) {
-    // --no-implicit-pubspec-resolution was passed, and synthetic-packages: true was not.
+  } else if (featureFlags.isExplicitPackageDependenciesEnabled) {
+    // synthetic-packages: true was not set and it is no longer the defualt.
     return;
   }
 
-  if (!environment.useImplicitPubspecResolution) {
+  if (featureFlags.isExplicitPackageDependenciesEnabled) {
     throwToolExit(
-      'Cannot generate a synthetic package when --no-implicit-pubspec-resolution is passed.\n'
+      'Cannot generate a synthetic package when --config explicit-package-dependencies is enabled.\n'
```

Comment:
Is this a typo? Is there `--config` option?

```suggestion
      'Cannot generate a synthetic package when explicit-package-dependencies is enabled.\n'
```